internal/k8s/ako_init.go

```diff
@@ -387,20 +387,26 @@ func (c *AviController) FullSyncK8s() error {
 	if err != nil {
 		utils.AviLog.Errorf("Unable to retrieve the services during full sync: %s", err)
 		return err
-	} else {
-		for _, svcObj := range svcObjs {
-			isSvcLb := isServiceLBType(svcObj)
-			var key string
-			if isSvcLb {
-				key = utils.L4LBService + "/" + utils.ObjKey(svcObj)
-			} else {
-				if lib.GetAdvancedL4() {
-					continue
-				}
-				key = utils.Service + "/" + utils.ObjKey(svcObj)
+	}
+	for _, svcObj := range svcObjs {
+
+		isSvcLb := isServiceLBType(svcObj)
+		var key string
+		if isSvcLb {
+			svcLabel := utils.ObjKey(svcObj)
+			ns := strings.Split(svcLabel, "/")
+			//Add key if ServiceAPI or advance L4 enabled or NS is valid
+			if !utils.IsServiceNSValid(ns[0]) {
```

Comment:
technically you can even stop the service of ClusterIP from getting processed because the namespace is not eligible right? I understand the ingress object won't be processed so we are good that way but this could be another optimisation.